### ISSUE TITLE
Can you provide the code to construct the synthesise annotations?

### ISSUE THREAD

**Issue by mrbeann** [createmomo/instance-level-reliability#1]
Can you please provide the code to construct the synthesize annotations?

**Comment by createmomo** [comment]
> Can you please provide the code to construct the synthesize annotations?

Hi mrbeann,

Many thanks for your interested in this work.

I am on vacations currently. But I will reply to your concern as early as possible.

Best wishes,
:)

**Comment by mrbeann** [comment]
Thanks for your reply. BTW, about the cross-entropy loss can you give some intuition of why this is a reasonable choice?

**Comment by createmomo** [comment]
> Can you please provide the code to construct the synthesize annotations?

Hi mrbeann,

Please find the related code for simulated annotations below:

**2-dimensional datasets:**
sklearn library is used for generating these 2d data.
`from sklearn.datasets.samples_generator import make_blobs, make_moons, make_circles`
and then:
`
X, y = make_moons(n_samples=1000, noise=0.1)`
`
X,y = make_circles(n_samples=1000, noise=0.05)`
`
X, y = make_blobs(n_samples = number_of_instances, centers=3, n_features=2, random_state=0)`

**Text classification datasets:** (pseudo-code)
As described in section 4.1, there are 4 types of simulated annotators:
*Adversarial*:
```
probability = np.random.uniform()
if probability >= 0.2: 
    this annotator's label = numpy.random.choice(the list contains all the possible labels except for the correct one) 
else: 
    this annotator's label = the correct label
```
*Random*:
```
this annotator's label = numpy.random.choice(the list contains all the possible labels)
```
*Broad*:
```
probability = np.random.uniform()
if probability <= 0.05:
    this annotator's label = numpy.random.choice(the list contains all the possible labels except for the correct one) 
else:
    this annotator's label = the correct label
```
*Narrow*:
```
probability = np.random.uniform()
if this annotator is good at labelling current label:
    this annotator's label = the correct label
else:
    if probability <= 0.35:
       this annotator's label = numpy.random.choice(the list contains all the possible labels except for the correct one)
   else:
       this annotator's label = the correct label
```
**Generating the annotations for Table 7:** (pseudo-code)
Example:
if the pre-defined probability of an annotator providing an incorrect label is 0.7:
for each instance and this annotator:
```
probability = np.random.uniform()
if probability <= 0.7:
     this annotator's label = numpy.random.choice(the list contains all the possible labels except for the correct one)
   else:
       this annotator's label = the correct label
```

**Comment by createmomo** [comment]
> Thanks for your reply. BTW, about the cross-entropy loss can you give some intuition of why this is a reasonable choice?

The intuition is from the k-means algorithm which alternates between 2 steps: assignment the predicted categories and re-calculate the centers.
